cmd/neofs-lens/internal/object/root.go

```diff
@@ -0,0 +1,278 @@
+	appCfg := config.New(config.Prm{}, config.WithConfigFile(vConfig))
+
+	ls := engine.New()
+
+	var shards []local_storage.ShardCfg
+	err := engineconfig.IterateShards(appCfg, false, func(sc *shardconfig.Config) error {
+		var sh local_storage.ShardCfg
+
+		sh.RefillMetabase = sc.RefillMetabase()
+		sh.Mode = sc.Mode()
+		sh.Compress = sc.Compress()
+		sh.UncompressableContentType = sc.UncompressableContentTypes()
+		sh.SmallSizeObjectLimit = sc.SmallSizeLimit()
+
+		// write-cache
+
+		writeCacheCfg := sc.WriteCache()
+		if writeCacheCfg.Enabled() {
+			wc := &sh.WritecacheCfg
+
+			wc.Enabled = true
+			wc.Path = writeCacheCfg.Path()
+			wc.MaxBatchSize = writeCacheCfg.BoltDB().MaxBatchSize()
+			wc.MaxBatchDelay = writeCacheCfg.BoltDB().MaxBatchDelay()
+			wc.MaxObjSize = writeCacheCfg.MaxObjectSize()
+			wc.SmallObjectSize = writeCacheCfg.SmallObjectSize()
+			wc.FlushWorkerCount = writeCacheCfg.WorkersNumber()
+			wc.SizeLimit = writeCacheCfg.SizeLimit()
+			wc.NoSync = writeCacheCfg.NoSync()
+		}
+
+		// blobstor with substorages
+
+		blobStorCfg := sc.BlobStor()
+		storagesCfg := blobStorCfg.Storages()
+		metabaseCfg := sc.Metabase()
+		gcCfg := sc.GC()
+
+		if config.BoolSafe(appCfg.Sub("tree"), "enabled") {
+			piloramaCfg := sc.Pilorama()
+			pr := &sh.PiloramaCfg
+
+			pr.Enabled = true
+			pr.Path = piloramaCfg.Path()
+			pr.Perm = piloramaCfg.Perm()
+			pr.NoSync = piloramaCfg.NoSync()
+			pr.MaxBatchSize = piloramaCfg.MaxBatchSize()
+			pr.MaxBatchDelay = piloramaCfg.MaxBatchDelay()
+		}
+
+		ss := make([]local_storage.SubStorageCfg, 0, len(storagesCfg))
+		for i := range storagesCfg {
+			var sCfg local_storage.SubStorageCfg
+
+			sCfg.Typ = storagesCfg[i].Type()
+			sCfg.Path = storagesCfg[i].Path()
+			sCfg.Perm = storagesCfg[i].Perm()
+
+			switch storagesCfg[i].Type() {
+			case blobovniczatree.Type:
+				sub := blobovniczaconfig.From((*config.Config)(storagesCfg[i]))
+
+				sCfg.Size = sub.Size()
+				sCfg.Depth = sub.ShallowDepth()
+				sCfg.Width = sub.ShallowWidth()
+				sCfg.OpenedCacheSize = sub.OpenedCacheSize()
+			case fstree.Type:
+				sub := fstreeconfig.From((*config.Config)(storagesCfg[i]))
+				sCfg.Depth = sub.Depth()
+				sCfg.NoSync = sub.NoSync()
+			case peapod.Type:
+				peapodCfg := peapodconfig.From((*config.Config)(storagesCfg[i]))
+				sCfg.FlushInterval = peapodCfg.FlushInterval()
+			default:
+				return fmt.Errorf("invalid storage type: %s", storagesCfg[i].Type())
+			}
+
+			ss = append(ss, sCfg)
+		}
+
+		sh.SubStorages = ss
+
+		// meta
+
+		m := &sh.MetaCfg
+
+		m.Path = metabaseCfg.Path()
+		m.Perm = metabaseCfg.BoltDB().Perm()
+		m.MaxBatchDelay = metabaseCfg.BoltDB().MaxBatchDelay()
+		m.MaxBatchSize = metabaseCfg.BoltDB().MaxBatchSize()
+
+		// GC
+
+		sh.GcCfg.RemoverBatchSize = gcCfg.RemoverBatchSize()
+		sh.GcCfg.RemoverSleepInterval = gcCfg.RemoverSleepInterval()
+
+		shards = append(shards, sh)
+
+		return nil
+	})
+	common.ExitOnErr(cmd, err)
```

Comment:
dont mind adding some details here

Reply:
as it returns in https://github.com/nspcc-dev/neofs-node/blob/feature/1336-neofs-lens-pull-object/cmd/neofs-lens/internal/storage/root.go#L137 is it enough of details?